astlib/config/dune

```diff
@@ -0,0 +1,4 @@
+(executable
+  (libraries ocaml-version)
+  (modules gen)
+  (name gen))
```

Comment:
This is interesting! I didn't know the `ocaml-version` library yet. In general, I think your suggestion is a good idea since it allows the basic `dune` workflow. However, fetching the ocaml version via the `dune` variable (as we're currently doing) has the advantage of keeping our dependency graph minimal. So, even though I like the simplicity of your suggestion, I'd prefer to keep it the way it is. What do you think?

Reply:
The problem here was (if I remember right) that a development version of the compiler has a very verbose version that wasn't parsed right.  No need to depend on ocaml_version externally -- if useful, it's just a single file you can pull into this repo.